src/snowflake/snowpark/row.py

```diff
@@ -48,19 +58,45 @@ class Row(tuple):
     >>> emp2
     Row(name='James', salary=20000)
 
+    We can optionally make accessing ``Row`` field attributes case insensitive:
+
+    >>> Employee = Row("FIRSTNAME", "SALARY", case_sensitive=False)
+    >>> emp1 = Employee("John", 10000)
+    >>> emp1.firstName
+    'John'
+    >>> emp2 = Employee("James", 20000)
+    >>> emp2.Salary
+    20000
+
     """
 
     def __new__(cls, *values: Any, **named_values: Any):
+        is_case_sensitive = True
         if values and named_values:
-            raise ValueError("Either values or named_values is required but not both.")
+            if len(named_values) == 1 and "case_sensitive" in named_values:
+                is_case_sensitive = named_values["case_sensitive"]
```

Comment:
what if case_sensitive is a column name? I guess you can make case_sensitive as an optional argument explicitly in `__new__`?

Reply:
Consider the case when we create a row like so: `lease_doc = Row(author="Name", legal=True, case_sensitive=True)`. From our side, it is not possible to decipher whether `case_sensitive` is a column in the row or if it is an argument

Reply:
I still think we should consider providing a factory/builder. This factory can explicitly set a row to be case sensitive or not, and also directly feed the preprocessed value to `Row` object. For example:

```
create_case_insensitive_row(original_args):
    values = [canonicalize_field(value) for value in values]
    row = Row(original_args)
    row.set_case_sensitive(False)
```

or 

```
builder = RowBuilder(args)
builder.set_case_sensitive(False)
row = builder.to_row()
```

I think we probably still should try to avoid making a specific string impossible to be a column name because of implementation.